File: cpp/bc2311.cpp
#include <iostream>
#include <sstream>
#include <vector>
#include <string>
#include <iomanip>

using namespace std;

int main()
{
	float med, res, biggest, smallest;
	vector <float> vec;
	int nCa;
	string name, line, token;
	
	cin >> nCa;
	cin.ignore();
	 
	for(int i = 0; i < nCa; i++)
	{
		res = 0;
		
		cin >> name;
		cin >> med;
		
		cin.ignore();
		getline(cin, line);
		
		istringstream iss(line);
		
		while(iss >> token)
			vec.push_back(stof(token));
		
		int tam = vec.size();
		biggest = vec[0];
		smallest = biggest;
		
		for(int j = 0; j < tam; j++)
		{
    	     if(vec[j] > biggest) biggest = vec[j];
    	     if(vec[j] < smallest) smallest = vec[j];
    	}
    	
    	for(int k = 0; k < tam; k++)
    	    res += vec[k];
    	   
    	res = (res - biggest - smallest) * med;
    	
		cout << name << " " << fixed << setprecision(2) << res << endl;
		vec.clear();
	}
}

File: python/bc2311.py
vec = [float]*7
nCa = int(input())

for c in range(0, nCa):
    res = 0

    name = input()
    med = float(input())

    line = input().split(" ")

    for d in range(0, 7):
        vec[d] = float(line[d])

    biggest = vec[0]
    smallest = biggest

    for e in range(0, 7):
        res += vec[e]
        if vec[e] > biggest: biggest = vec[e]
        if vec[e] < smallest: smallest = vec[e]

    res = (res - biggest - smallest) * med

    print(f"{name} {res:.2f}")
